applications/StructuralMechanicsApplication/python_scripts/set_automated_initial_variable_process.py

```diff
@@ -0,0 +1,94 @@
+        raise RuntimeError(ErrorMsg)
+    else:
+        component_list = []
+        table_id_list = []
+        out_of_range_component_list = []
+        for i in range (0, len(layer_list)):
+            component_number = int(layer_list[i].stem.split("_")[1][-1])
+            if  0 < component_number < 7:
+                component_list.append(component_number)
+                table_id = int(layer_name[-1] + str(component_number - 1))
+                table_id_list.append(table_id)
+                process_settings["initial_variable_table"]["filename"].SetString(layer_list[i].as_posix())
+                process_settings["initial_variable_table"]["table_id"].SetInt(table_id)
+                ReadCsvTableUtility(process_settings["initial_variable_table"]).Read(computing_model_part)
+            else:
+                out_of_range_component_list.append(component_number)
+
+    raw_variable_name = process_settings["variable_name"].GetString()
+    variable_name = raw_variable_name.split("_")[0] + " " + raw_variable_name.split("_")[1].split("_")[0]
+
+    if len(out_of_range_component_list) != 0:
+        if len(out_of_range_component_list) == 1:
+            Logger.PrintWarning("SetAutomatedInitialVariableProcess:: ", f"{variable_name.capitalize()} component {out_of_range_component_list[0]} of {layer_name} is out of range. The corresponding table will be neglected")
+        else:
+            out_of_range_component_name = ", ".join(str(out_of_range_component) for (out_of_range_component) in out_of_range_component_list)
+            Logger.PrintWarning("SetAutomatedInitialVariableProcess:: ", f"{variable_name.capitalize()} components {out_of_range_component_name} of {layer_name} are out of range. Correspoding tables will be negleted")   
+
+    if len(component_list) < 6:
+        missing_component_list = list(set(range(1,7)).difference(component_list))
+        if len(missing_component_list) == 1:
+            Logger.PrintWarning("SetAutomatedInitialVariableProcess:: ", f"Table correspoding to {variable_name.lower()} component {str(missing_component_list[0])} of {layer_name} not found. A zero entry will be added to the {raw_variable_name} variable")
+        else:
+            missing_component_name = ", ".join(str(missing_component) for (missing_component) in missing_component_list)
+            Logger.PrintWarning("SetAutomatedInitialVariableProcess:: ", f"Tables correspoding to {variable_name.lower()} components {missing_component_name} of {layer_name} not found. Zero entries will be added to the {raw_variable_name} variable")
+    else:
+        Logger.PrintInfo("SetAutomatedInitialVariableProcess:: ", f"{variable_name.capitalize()} tables of {layer_name} were successfully imported")
+    
+    default_table_id_vector = KM.Parameters("""{
+    "table_id_vector": [10,11,12,13,14,15]
+    }""")
+    process_settings.AddValue("table_id_vector", default_table_id_vector)
+    process_settings["table_id_vector"].SetVector(table_id_list)
```

Comment:
```suggestion
    process_settings.AddEmptyValue("table_id_vector").SetVector(table_id_list)
```

sorry this should work